dydxV4/dydxV4/_Notification/dydxNotificationHandlerDelegate.swift

```diff
@@ -66,71 +69,30 @@ final class dydxNotificationHandlerDelegate: NSObject, NotificationHandlerDelega
     }
 
-//    private func sendTokenUpdate() {
-//        if let token = token, let ethereumAddress = walletConnectionInteractor?.walletConnection?.ethereumAddress,
-//           permission == .authorized {
-//            sendNotificationTokenUpdate(token: token, ethereumAddress: ethereumAddress)
-//        }
-//    }
-//
-//    private func sendTokenDeletion() {
-//        if let token = token, let ethereumAddress = walletConnectionInteractor?.walletConnection?.ethereumAddress {
-//            sendNotificationTokenDeletion(token: token, ethereumAddress: ethereumAddress)
-//        }
-//    }
-//
-//    private func didSetWalletConnectionInteractor(oldValue: dydxWalletConnectionInteractor?) {
-//        // send token update or deletion when user sign-in or sign-out
-//        if let token = token {
-//            if let ethereumAddress = walletConnectionInteractor?.walletConnection?.ethereumAddress, permission == .authorized {
-//                sendNotificationTokenUpdate(token: token, ethereumAddress: ethereumAddress)
-//            } else if let ethereumAddress = oldValue?.walletConnection?.ethereumAddress {
-//                sendNotificationTokenDeletion(token: token, ethereumAddress: ethereumAddress)
-//            }
-//        }
-//    }
-//
-//    private var tokenApi: dydxPrivateApi?
-//
-//    private func sendNotificationTokenUpdate(token: String, ethereumAddress: String) {
-//        if dydxBoolFeatureFlag.push_notification.isEnabled,
-//           let path = dydxPrivateApi.endpointResolver.path(for: "registration-tokens") {
-//            let api = dydxPrivateApi(ethereumAddress: ethereumAddress)
-//            api.post(path: path, params: nil, data: ["token": token]) { [weak self] _, error in
-//                if let error = error {
-//                    Console.shared.log("Firebase token registration failed: \(error.localizedDescription)")
-//                }
-//                self?.tokenApi = nil
-//            }
-//            tokenApi = api
-//        }
-//    }
-//
-//    private func sendNotificationTokenDeletion(token: String, ethereumAddress: String) {
-//        if dydxBoolFeatureFlag.push_notification.isEnabled,
-//           let path = dydxPrivateApi.endpointResolver.path(for: "registration-tokens"),
-//           let encodedToken = token.encodeBase64() {
-//            let api = dydxPrivateApi(ethereumAddress: ethereumAddress)
-//            api.delete(path: path + "/" + encodedToken, params: nil) { [weak self] _, error in
-//                if let error = error {
-//                    Console.shared.log("Firebase token deletion failed: \(error.localizedDescription)")
-//                }
-//                self?.tokenApi = nil
-//            }
-//            tokenApi = api
-//        }
-//    }
+    private func sendTokenUpdate() {
+        if let token = token, permission == .authorized {
+            AbacusStateManager.shared.registerPushNotification(token: token, languageCode: languageCode)
+        }
```

Comment:
even if user is not authorized, I think we would still want to send the language/token update in case they authorize it in the background

Reply:
That would mean the backend will send notifications to all users.  When the notification is enabled from the device Settings, this will get called when the app is foregrounded again, so it should handle most cases.